extension/core_functions/aggregate/distributive/max_intersections.cpp

```diff
@@ -0,0 +1,203 @@
+#include "core_functions/aggregate/distributive_functions.hpp"
+#include <algorithm>
+#include <vector>
+#include "vergesort.h"
+
+#ifndef DUCKDB_NO_THREADS
+#include "duckdb/common/thread.hpp"
+#endif
+
+namespace duckdb {
+
+namespace {
+
+struct MaxIntersectionsState {
+	std::vector<int64_t> starts;
+	std::vector<int64_t> ends;
+
+	MaxIntersectionsState() {
+		starts.reserve(64);
+		ends.reserve(64);
+	}
+
+	MaxIntersectionsState(MaxIntersectionsState &&other) noexcept
+	    : starts(std::move(other.starts)), ends(std::move(other.ends)) {
+	}
+
+	MaxIntersectionsState &operator=(MaxIntersectionsState &&other) noexcept {
+		if (this != &other) {
+			starts = std::move(other.starts);
+			ends = std::move(other.ends);
+		}
+		return *this;
+	}
+
+	MaxIntersectionsState(const MaxIntersectionsState &) = delete;
+	MaxIntersectionsState &operator=(const MaxIntersectionsState &) = delete;
+};
+
+struct MaxIntersectionsFunction {
+	template <class STATE>
+	static void Initialize(STATE &state) {
+		new (&state) MaxIntersectionsState();
+	}
+
+	template <class STATE>
+	static void Destroy(STATE &state, AggregateInputData &aggr_input_data) {
+		state.~MaxIntersectionsState();
+	}
+
+	template <class A_TYPE, class B_TYPE, class STATE, class OP>
+	static void Operation(STATE &state, const A_TYPE &left, const B_TYPE &right, AggregateBinaryInput &input) {
+		// Validate interval and add only if valid (left <= right)
+		if (left <= right) {
+			state.starts.emplace_back(static_cast<int64_t>(left));
+			state.ends.emplace_back(static_cast<int64_t>(right) +
+			                        1); // Use exclusive end for correct overlap calculation
+		}
+	}
+
+	template <class A_TYPE, class B_TYPE, class STATE, class OP>
+	static void ConstantOperation(STATE &state, const A_TYPE &left, const B_TYPE &right, AggregateBinaryInput &input,
+	                              idx_t count) {
+		// For constant operations, add the interval once per count if valid
+		if (left <= right) {
+			state.starts.reserve(state.starts.size() + count);
+			state.ends.reserve(state.ends.size() + count);
+
+			int64_t start_val = static_cast<int64_t>(left);
+			int64_t end_val = static_cast<int64_t>(right) + 1;
+
+			for (idx_t i = 0; i < count; i++) {
+				state.starts.emplace_back(start_val);
+				state.ends.emplace_back(end_val);
+			}
+		}
+	}
+
+	template <class STATE, class OP>
+	static void Combine(const STATE &source, STATE &target, AggregateInputData &) {
+		if (source.starts.empty()) {
+			return;
+		}
+
+		target.starts.reserve(target.starts.size() + source.starts.size());
```

Comment:
This will copy the data - which is great for windowing, but if you can move the data when `AggregateInputData .combine_type == AggregateCombineType::ALLOW_DESTRUCTIVE` you will reduce the memory footprint during normal aggregation.